src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/LSPDocument.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    public abstract class LSPDocument
```

Comment:
>LSPDocument [](start = 26, length = 11)

Nit: I noticed you don't use interfaces anywhere, preferring pure abstract classes. Any reason why?

Reply:
Can make changes to them without being a binary breaking change 😄. Well that's the old argument anyways lol, now that in C# 8.0 you can have default interface implementations that argument no longer holds true.

My core reason now for using abstract classes instead of interfaces is that an interface describes an "I can do XYZ" relationship where an abstract class is more of an "I am XYZ" relationship.